3rdParty/fuerte/src/vst.cpp

```diff
@@ -455,27 +467,34 @@ RequestHeader requestHeaderFromSlice(VPackSlice const& headerSlice) {
   header.byteSize = headerSlice.byteSize();  // for debugging
 #endif
 
-  header.setVersion( headerSlice.at(0).getNumber<short>()); // version
-  assert(headerSlice.at(1).getNumber<int>() == static_cast<int>(MessageType::Request));
+  header.setVersion(headerSlice.at(0).getNumber<short>());  // version
+  assert(headerSlice.at(1).getNumber<int>() ==
+         static_cast<int>(MessageType::Request));
   header.database = headerSlice.at(2).copyString();  // databse
-  header.restVerb = static_cast<RestVerb>(headerSlice.at(3).getInt());  // rest verb
+  header.restVerb =
+      static_cast<RestVerb>(headerSlice.at(3).getInt());  // rest verb
   header.path = headerSlice.at(4).copyString();           // request (path)
-  header.parameters = sliceToStringMap(headerSlice.at(5));  // query params
-  header.meta = sliceToStringMap(headerSlice.at(6));        // meta
-
+  for (auto const& it :
+       VPackObjectIterator(headerSlice.at(5))) {  // query params
+    header.parameters.emplace(it.key.copyString(), it.value.copyString());
+  }
+  for (auto const& it :
+       VPackObjectIterator(headerSlice.at(5))) {  // meta (headers)
```

Comment:
`headerSlice.at(5)` is added to both `header.parameters` and to `header.addMeta`. Is that intentional?
```suggestion
       VPackObjectIterator(headerSlice.at(6))) {  // meta (headers)
```